### ISSUE TITLE
When missing_keys source is missing, a KeyMappingError should be raised, not a MissingHeaders exception!

### ISSUE THREAD

**Issue by KevinColemanInc** [tilo/smarter_csv#139]
If you have a key mapping hash, like:

```
{ name: full_name }
```

Then you would expect the header of "name" to be mapped to "full_name", but if "name" is missing, the exception thrown says you are missing "full_name"

=> "ERROR: missing headers: full_name"

Impacted version:
v1

**Comment by tilo** [comment]
This is raised by v 1.2.x, here:
https://github.com/tilo/smarter_csv/blob/4b192ee76a0a39fe3db2b5332614dc0d3664be34/lib/smarter_csv/smarter_csv.rb#L111

but that only happens when you define `required_headers`

**Comment by tilo** [comment]
@KevinColemanInc I can't reproduce this.

```
data = SmarterCSV.process('/tmp/test.csv', {key_mapping_hash: {user_id: 'User ID', employer_name: 'Employer Name', missing: 'Missing'}})
=> correctly parsed data, no exception
```

can you provide an example for this?

**Comment by KevinColemanInc** [comment]
You tell your user to upload a CSV file with 1 column with the title "manager_email", which you will map to "parent_email".

Your code would look like this:
```
options = { :required_headers => [:parent_email],
            :key_mapping => {manager_email: :parent_email}
          } # order does not matter
SmarterCSV.process("#{fixture_path}/user_import.csv", options)
```
Lets say the user uploads a CSV with the header of "email" instead of "manager_email".

SmarterCSV will throw: `ERROR: missing headers: parent_email`. This is confusing, because we told the user the "correct" column should be `manager_email`, not "parent_email".

Another example:
You can add this failing test to `invalid_headers_spec.rb`
```
 it 'raises an error if a required header is missing' do
    human_column_name = :middlename
    expect {
      options = { :required_headers => [:middle_name],
                  :key_mapping => { human_column_name => :middle_name}
                } # order does not matter
      SmarterCSV.process("#{fixture_path}/user_import.csv", options)
    }.to raise_exception(SmarterCSV::MissingHeaders, "ERROR: missing headers: #{human_column_name}")
  end
```
I am thinking SmarterCSV should return the "human" name in the exception, instead of "code" name.

**Comment by tilo** [comment]
the problem here is that SmarterCSV prints a warning for the missing header, instead of raising an exception

this needs to be fixed in 1.9.0 (will be a breaking change)

**Comment by tilo** [comment]
ALSO: need to rename `MissingHeaders` exception to `MissingKeys`

**Comment by tilo** [comment]
@KevinColemanInc this is addressed in this PR: https://github.com/tilo/smarter_csv/pull/242

**Comment by tilo** [comment]
fixed in version 1.9.0
